### PR TITLE
feat: add bounded YARA-X rule validation

### LINKED ISSUE #170: Guardian: replace heuristic YARA checks with bounded YARA-X validation
## Acceptance Criteria
- [ ] Candidate YARA is validated by a pinned real compiler, not substring heuristics
- [ ] Validation is compile-only, in-memory, bounded, and fail-closed
- [ ] Includes, imports/modules, NUL/non-ASCII input, oversized input, warnings, and multiple-rule payloads are rejected
- [ ] Existing analyzer integration keeps its interface and passing validation grants no publication, submission, v2, wallet, chain, reward, deployment, or production authority
- [ ] Adversarial tests cover comment-smuggling, invalid syntax, valid stringless rules, includes/imports, globals, warnings, duplicate/multiple rules, bounds, and malformed metadata
- [ ] Full Guardian and repository verification passes; Bridge/Memory/README/Whitepaper/Pages stay synchronized

## Context
GH-167 closed the bounded ThreatHint-v2 repository transport substrate. The next repository-owned prerequisite for privacy-reviewed semantic/actionable analysis is real rule validation. The current Guardian validator only searches for `rule `, `strings:`, and `condition:` substrings, which can accept invalid source and rejects valid YARA without a strings section.

## Proposed Approach
Add a dedicated fail-closed YARA-X candidate validator using an exact pinned official dependency. Compile source from memory with includes disabled, reject any imports/modules and warnings, enforce strict source/rule-name/metadata bounds, and delegate the existing `YaraRuleGenerator.validate_rule` to it. Keep all scanning, durable semantic output, v2 worker integration, publication, committee, wallet, chain, and deployment explicitly out of scope.

## Risk
Medium: security validation boundary and native dependency. Requires adversarial tests, dependency audit, independent review, and complete protected CI.

### PR DESCRIPTION
**Pull request by NeaBouli**
## Summary
- replace substring-only Guardian candidate-rule checks with exact-pinned compile-only YARA-X validation
- reject imports/includes, multiple rules, malformed or warning-producing source under explicit input budgets
- add adversarial coverage and synchronize README, Whitepaper, Roadmap, FAQ/Pages, protocol, Memory, and Bridge

## Security boundary
- in-memory compile only; no file/process/data scan
- no model invocation, actionable v2 result, submission, publication, wallet, chain, deployment, or production authority

## Verification
- 76 focused tests; full Guardian: 1207 passed, 4 skipped
- Black clean; changed-source Pylint 10.00/10; full Guardian Pylint 9.85/10
- Pip Audit: no known vulnerabilities
- Rust workspace fmt/clippy/tests; locked release binaries/packages; release performance gate
- Memory/Autodidactic, H-001 public evidence, HTML/JSON-LD/YAML/stale-status, Cargo Audit and diff hygiene
- independent Kimi security review: no P0/P1/P2

Local Docker is unavailable, so protected Linux CI is authoritative for the Compose render step; the structural vLLM policy verifier passes.

Closes #170

## Summary by CodeRabbit

- **New Features**
  - Added bounded YARA rule validation using pinned YARA-X 1.4.0.
  - Validates a single ASCII rule within size and naming limits.
  - Rejects imports, includes, multiple rules, unsupported content, compiler errors, and warnings.
  - Performs compile-only checks without scanning data or enabling production actions.

- **Documentation**
  - Added roadmap, FAQ, whitepaper, audit, and status updates describing the validation boundary and pending merge and protected CI verification.

- **Tests**
  - Added coverage for valid rules, malformed input, parser edge cases, and rejected constructs.

### CODE REVIEW

File: docs/agent-bridge/ACTION_LOG.md [hunk 1]
```diff
@@ -3510,3 +3510,40 @@ Rules for all dev agents:
...
+
+- Opened issue #170 and branch `feat/GH-170-yara-x-validation` from clean exact
+  final main `e8a13cae679a4d1a5d213e0d56b9e958dfa20733`.
+- Bounded scope is compile-only in-memory candidate validation with official
+  pinned YARA-X, strict source/feature budgets, adversarial tests, and no scan or
+  external effect. Existing v2 non-actionable worker and durable schema remain
+  unchanged.
+- Kimi supplied read-only architecture analysis; Sol owns implementation review,
+  dependency/security decisions, full gates, publication, and closeout.
+- Status: `In Progress / Implementation and dependency verification next`.
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply docs/agent-bridge/ACTION_LOG.md:3524]: No change needed: the audit timezone is explicitly Europe/Athens (EEST). GitHub submitted this review on 2026-08-12 UTC, which is 2026-08-13 EEST; the recorded 01:31-01:56 EEST milestones are therefore contemporaneous, not future-dated.

File: memory/CHECKPOINT.md [hunk 2]
```diff
@@ -14,11 +14,11 @@
...
-| Aktueller HEAD | Canonical product/public main is `7c6260855193cce1ae2790670fcf25371ac08412`; the documentation-only GH-167 closeout branch records that exact-main evidence |
+| Aktueller HEAD | Branch `feat/GH-170-yara-x-validation` is based on exact final main `e8a13cae679a4d1a5d213e0d56b9e958dfa20733`; GH-170 is local-only until protected merge |
 | Rollback-Tag | `pre-session-20260413` → `6347b85` |
-| Whitepaper | WHITEPAPER.md (root) + whitepaper.html (styled); miner companion, reward boundaries, GH-36/GH-39 trust boundaries, GH-48/GH-52/GH-55/GH-58/GH-63/GH-74/GH-94/GH-107/GH-111/GH-114, merged GH-117 binding/verifier/preflight/acceptance/governance/outbox/non-actionable-worker, and GH-167 bounded transport boundaries remain synchronized |
+| Whitepaper | WHITEPAPER.md (root) + whitepaper.html (styled); GH-167 merged transport and GH-170 local compile-only YARA-X boundaries are synchronized without claiming semantic or production authority |
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply memory/CHECKPOINT.md:21]: Fixed in e114410: the retained GH-167 section and its captured status are now explicitly labeled historical; the current checkpoint already names GH-170 as the active slice.

File: modules/guardian-node/tests/test_yara_validation.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            _simple_rule("Tagged : alpha beta"),
+            "private " + _simple_rule("P"),
+            "global " + _simple_rule("G"),
+            "private global " + _simple_rule("PG"),
+            _simple_rule("Re", "$a = /a.*b/"),
+            _simple_rule("Quant", "$a = /ab{2,3}c/"),
+            _simple_rule("Klass", "$a = /[a-z\\/]{4,}/"),
+            _simple_rule("Jumps", "$a = { 4D 5A [0-4] 90 ?? }"),
+            _simple_rule("N" * MAX_YARA_RULE_NAME_LENGTH),
+        ],
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply modules/guardian-node/tests/test_yara_validation.py:57]: Fixed in e114410: added the valid stringless `filesize > 0` case and reran the focused boundary suite (77 passed). Memory Integrity, Black, and diff hygiene also pass.
